src/PackageSelection/PackageSelection.php

```diff
@@ -180,34 +179,37 @@ public function select(Composer $composer, bool $verbose): array
             }
         }
 
-        $this->addRepositories($pool, $repos);
+        $repositorySet = new RepositorySet($this->minimumStability, $stabilityFlags);
+        $this->addRepositories($repositorySet, $repos);
 
         // determine the required packages
         $rootLinks = $this->requireAll ? $this->getAllLinks($repos, $this->minimumStability, $verbose) : $this->getFilteredLinks($composer);
 
         // select the required packages and determine dependencies
-        $depsLinks = $this->selectLinks($pool, $rootLinks, true, $verbose);
+        $depsLinks = $this->selectLinks($repositorySet, $rootLinks, true, $verbose);
 
         if ($this->requireDependencies || $this->requireDevDependencies) {
+            $repositorySet = new RepositorySet($this->minimumStability, $stabilityFlags);
```

Comment:
do we need to create a new one here ?

Reply:
Yes, when a `Pool` has been created by the `RepositorySet` ([in `selectLinks`](https://github.com/composer/satis/pull/645/files#diff-d16a5020dc605f5cb12e0cf898bb6a7d2b2a7eacd2dc2bbc77b56b1d61d005a3R721)), it is locked. Then other repositories [cannot be added](https://github.com/composer/composer/blob/2.0.12/src/Composer/Repository/RepositorySet.php#L128).